src/sqlfluff/core/parser/segments/base.py

```diff
@@ -391,6 +423,38 @@ def raw(self) -> str:
         """Make a string from the segments of this segment."""
         return "".join(seg.raw for seg in self.segments)
 
+    @cached_property
```

Comment:
There's probably no reason to cache this, since it's simply returning an already-computed set.

I think could even eliminate this function completely by changing the metaclass to populate `class_types` rather than `_class_types`.

Reply:
`RawSegment` overrides `class_types` to allow it to include the instance variable `_surrogate_type`. Is still needs the class method, but this shim is there to be overridden. You're probably right that it doesn't need to be cached though.